python/cudf/cudf/core/indexed_frame.py

```diff
@@ -2649,6 +2697,705 @@ def _explode(self, explode_column: Any, ignore_index: bool):
             res.index.names = self._index.names
         return res
 
+    @_cudf_nvtx_annotate
+    @docutils.doc_apply(
+        doc_binop_template.format(
+            operation="Addition",
+            op_name="add",
+            equivalent_op="frame + other",
+            df_op_example=""">>> df.add(1)
+                   angles  degrees
+        circle          1      361
+        triangle        4      181
+        rectangle       5      361
+            """,
+            ser_op_example=""">>> a.add(b)
+        a       2
+        b    <NA>
+        c    <NA>
+        d    <NA>
+        e    <NA>
+        dtype: int64
+        >>> a.add(b, fill_value=0)
+        a       2
+        b       1
+        c       1
+        d       1
+        e    <NA>
+        dtype: int64""",
+        )
+    )
+    def add(self, other, axis, level=None, fill_value=None):  # noqa: D102
+        if level is not None:
+            raise NotImplementedError("level parameter is not supported yet.")
+
+        return self._binaryop(other, "__add__", fill_value)
+
+    @_cudf_nvtx_annotate
+    @docutils.doc_apply(
+        doc_binop_template.format(
+            operation="Addition",
+            op_name="add",
+            equivalent_op="other + frame",
+            df_op_example=""">>> df.add(1)
```

Comment:
Any reason for the switch to the `add` method instead of the binary operator? (This question also applies for all the other binops docstrings)

```suggestion
            df_op_example=""">>> df + 1
```

Just want to make sure that the binops methods have their corresponding operators documented - if this usage is documented somewhere else then this change should be fine 🙂

Reply:
These docstrings are specifically the ones that will show up when look up `help(df.add)`, so we want the example to show how `df.add(1)` behaves, not `df + 1`. There isn't really a great way to document operators in docstrings because that documentation won't show up anywhere automatically. A user would have to do something awkward like `help(df.__add__)`, and very few packages actually populate those methods. We're better off showing examples of binary operators.

To be clear, this isn't a change; other than this typo for `add`, all (or at least most, maybe there were a few other typos) the other previously existing docstrings were for the methods, not the operators.